libs/client/cornflow_client/constants.py

```diff
@@ -90,6 +90,93 @@
     pl.LpStatusUndefined: STATUS_UNDEFINED,
 }
 
+PARAMETER_SOLVER_TRANSLATING_MAPPING = {
+    ('bar_iter_limit', 'pyomo', 'cbc') : 'barr' ,
+    ('best_obj_stop', 'pyomo', 'cbc') : 'primalT' ,
+    ('cutoff', 'pyomo', 'cbc') : 'cuts' ,
+    ('iteration_limit', 'pyomo', 'cbc') : 'maxIt' ,
+    ('solution_limit', 'pyomo', 'cbc') : 'maxSaved' ,
+    ('time_limit', 'pyomo', 'cbc') : 'sec' ,
+    ('mip_gap', 'pyomo', 'cbc') : 'allow' ,
+    ('optimality_tol', 'pyomo', 'cbc') : 'dualT' ,
+    ('pump_passes', 'pyomo', 'cbc') : 'pumpC' ,
+    ('heuristics', 'pyomo', 'cbc') : 'heur' ,
+
+    ('is_mip', 'pulp', 'cbc') : 'mip' ,
+    ('optimality_tol', 'pulp', 'cbc') : 'gapAbs' ,
+    ('cutoff', 'pulp', 'cbc') : 'cuts' ,
+    ('time_limit', 'pulp', 'cbc') : 'timeLimit' ,
+    ('threads', 'pulp', 'cbc') : 'threads' ,
+    ('presolve', 'pulp', 'cbc') : 'presolve' ,
+    
+
+    ('bar_iter_limit', 'pyomo', 'gurobi') : 'BarIterLimit' ,
+    ('best_obj_stop', 'pyomo', 'gurobi') : 'BestObjStop' ,
+    ('cutoff', 'pyomo', 'gurobi') : 'Cutoff' ,
+    ('iteration_limit', 'pyomo', 'gurobi') : 'IterationLimit' ,
+    ('mem_limit', 'pyomo', 'gurobi') : 'MemLimit' ,
+    ('solution_limit', 'pyomo', 'gurobi') : 'SolutionLimit' ,
+    ('time_limit', 'pyomo', 'gurobi') : 'TimeLimit' ,
+    ('mip_gap', 'pyomo', 'gurobi') : 'MIPGap' ,
+    ('optimality_tol', 'pyomo', 'gurobi') : 'OptimalityTol' ,
+    ('feasibility_tol', 'pyomo', 'gurobi') : 'FeasibilityTol' ,
+    ('branch_dir', 'pyomo', 'gurobi') : 'BranchDir' ,
+    ('pump_passes', 'pyomo', 'gurobi') : 'PumpPasses' ,
+    ('heuristics', 'pyomo', 'gurobi') : 'Heuristics' ,
+    ('presolve', 'pyomo', 'gurobi') : 'Presolve' ,
+    ('threads', 'pyomo', 'gurobi') : 'threads' ,
+    
+    ('is_mip', 'pulp', 'gurobi') : 'mip' ,
+    ('time_limit', 'pulp', 'gurobi') : 'timeLimit' ,
+    ('presolve', 'pulp', 'gurobi') : 'presolve' ,
+    ('mip_gap', 'pulp', 'gurobi') : 'MIPGap' ,
+    ('optimality_tol', 'pulp', 'gurobi') : 'gapAbs' ,
+    ('feasibility_tol', 'pulp', 'gurobi') : 'FeasibilityTol' ,
+    ('iteration_limit', 'pulp', 'gurobi') : 'IterationLimit' ,
```

Comment:
It is not clear to me it allows more parameters. Let me answer you first the question
_, why is it that ('optimality_tol', 'pulp', 'gurobi') : 'gapAbs' and ('optimality_tol', 'pyomo', 'gurobi') : 'OptimalityTol' ?_
pyomo and pulp have different api. Therefore, same prameters can (and do) have diffferent names. Regarding your point of the different number of passed parameters, actually, pulp does not accept the same parameters as pyomo. Pulp has a keyword "option" that accpets a list of options (which is different from the **kwargs parameter discussed in some answers above). Looking into the pulp api, it seems that this argument option is not  accepting a lot of new parameters (https://github.com/coin-or/pulp/blob/master/pulp/apis/coin_api.py line 23)

Reply:
Sorry for intruding the conversation. I am receiving mails about this PR (for being subscribed to cornflow) and I was intrigued by this.
The easier way to know what pulp is doing with the parameters is looking at its [code](https://github.com/coin-or/pulp/blob/master/pulp/apis/gurobi_api.py).

pulp arguments seems to be a little inconsistents depending on the solvers and the interface (python or cmd lines) but for gurobi, gapAbs does not seem to be mapped to OptimalityTol:
```
             logPath="LogFile",
            gapRel="MIPGap",
            gapAbs="MIPGapAbs",
            threads="Threads",
```
The former is in the cmd line class. The python class does not seems to have a gapAbs argument.
However the python Gurobi class has a **kwargs argument: `**solverParams`
So it would need to be tested but I think you could actually map pukp exactly like pyomo for Gurobi.

As I said, pulp is not perfectly consistent and the cbc class does not seems to have a kwarg.

Reply:
Also, there seems to be a confusion here between absolute gap and optimality tolerance for gurobi which are not the same parameter:
[OptimalityTol](https://www.gurobi.com/documentation/current/refman/optimalitytol.html#parameter:OptimalityTol)
[MIPGapAbs](https://www.gurobi.com/documentation/current/refman/mipgapabs.html#parameter:MIPGapAbs)

Personally, I have never used OptimalityTol

Reply:
You are more than welcome, Hugo! Indeed, you are right and we discussed this issue in an internal call. In this link https://github.com/coin-or/pulp/blob/master/pulp/apis/gurobi_api.py (line 336) it is the map I wrote (CMD-based). I took quite a few of these arguments from the **kwargs, and I assumed that they are gurobi-liked spelled

Reply:
Ah yes, if I am not mistaken the difference between python and cmd interface is that you need gurobipy installed for python interface, right?
So yes, you need to use the cmd interface and it does not have a kwarg.

Reply:
In any case, it might be a good idea to raise an issue in pulp in order to ask for a kwargs in all solver options and to ask them to review the consistency of arguments between interfaces.

Reply:
Thanks for pointing out the confusion on the MIPGapAbs and OptimalityTol. It will be corrected in the new commit

Reply:
What about for the people that do have gurobipy installed and are using the python interface?